pkgs/native_assets_cli/lib/src/api/asset.dart

```diff
@@ -2,58 +2,54 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import '../model/asset.dart' as model;
-import '../model/link_mode.dart' as model;
-import '../model/target.dart' as model;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart';
+
+import '../model/target.dart';
+import '../utils/map.dart';
+import '../utils/yaml.dart';
+import 'architecture.dart';
+import 'build_config.dart';
+import 'build_output.dart';
 import 'link_mode.dart';
-import 'target.dart';
+import 'os.dart';
 
-abstract class AssetPath {}
+part 'c_code_asset.dart';
+part 'data_asset.dart';
+part '../model/asset.dart';
+part '../model/c_code_asset.dart';
+part '../model/data_asset.dart';
 
-/// Asset at absolute path [uri].
-abstract class AssetAbsolutePath implements AssetPath {
-  Uri get uri;
-
-  factory AssetAbsolutePath(Uri uri) = model.AssetAbsolutePath;
-}
-
-/// Asset is avaliable on the system `PATH`.
+/// Data or code bundled with a Dart or Flutter application.
 ///
-/// [uri] only contains a file name.
-abstract class AssetSystemPath implements AssetPath {
-  Uri get uri;
-
-  factory AssetSystemPath(Uri uri) = model.AssetSystemPath;
-}
-
-/// Asset is loaded in the process and symbols are available through
-/// `DynamicLibrary.process()`.
-abstract class AssetInProcess implements AssetPath {
-  factory AssetInProcess() = model.AssetInProcess;
-}
-
-/// Asset is embedded in executable and symbols are available through
-/// `DynamicLibrary.executable()`.
-abstract class AssetInExecutable implements AssetPath {
-  factory AssetInExecutable() = model.AssetInExecutable;
-}
-
-abstract class Asset {
-  LinkMode get linkMode;
+/// An asset is data or code which is accessible from a Dart or Flutter
+/// application. To access an asset at runtime, the asset [id] is used. This
+/// enables access to the asset irrespective of how and where the application is
+/// run.
+abstract final class Asset {
+  /// The identifier for this asset.
+  ///
+  /// An [Asset] must have a string identifier called "asset id". Dart code that
+  /// uses an asset, references the asset using this asset id.
+  ///
+  /// A package must prefix all asset ids it defines with:
+  /// `package:<package>/`, `<package>` being the current package's name. This
+  /// ensures assets don't conflict between packages.
+  ///
+  /// Additionally, the convention is that an asset referenced from
+  /// `lib/src/foo.dart` in `package:foo` has the asset id
+  /// `'package:foo/src/foo.dart'`.
```

Comment:
Isn't it more accurate to say that: "the default assetId for an asset referenced from.... is ...".

You can reference other assetIds, if you really want to. But if you don't specify one, they this is the default one, which makes your life easy :D